src/metabase/warehouses/api.clj

```diff
@@ -983,6 +986,9 @@
        ;; with the advanced-config feature enabled.
         (when (premium-features/enable-cache-granular-controls?)
           (t2/update! :model/Database id {:cache_ttl cache_ttl}))
+        ;; provider_name should be updatable to nil when explicitly provided in request body
```

Comment:
this should not be a separate db update. It should be in the one update above. Put it outside of the `m/remove-vals nil?` call and use the top level merge to do something like `(when-let [[_ provider (find body :provider_name)] {:provider_name provider_name})`. We shouldn't be checking later and transacting again.

Reply:
Right! Thanks!

Fixed: https://github.com/metabase/metabase/pull/62057/commits/1a215d8dd7d48f76bd3c22014065c3796758073a